server/src/main/java/io/crate/expression/operator/all/AllEqOperator.java

```diff
@@ -0,0 +1,122 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.expression.operator.all;
+
+import static io.crate.metadata.functions.TypeVariableConstraint.typeVariable;
+
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.function.IntPredicate;
+import java.util.stream.StreamSupport;
+
+import org.apache.lucene.search.BooleanClause;
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.MatchNoDocsQuery;
+import org.apache.lucene.search.Query;
+import org.elasticsearch.common.lucene.search.Queries;
+import org.jetbrains.annotations.Nullable;
+
+import io.crate.expression.operator.AllOperator;
+import io.crate.expression.operator.EqOperator;
+import io.crate.expression.operator.Operator;
+import io.crate.expression.operator.any.AnyNeqOperator;
+import io.crate.expression.predicate.IsNullPredicate;
+import io.crate.expression.scalar.ArrayUnnestFunction;
+import io.crate.expression.symbol.Function;
+import io.crate.expression.symbol.Literal;
+import io.crate.expression.symbol.Symbol;
+import io.crate.lucene.LuceneQueryBuilder;
+import io.crate.metadata.FunctionType;
+import io.crate.metadata.Reference;
+import io.crate.metadata.functions.BoundSignature;
+import io.crate.metadata.functions.Signature;
+import io.crate.sql.tree.ComparisonExpression;
+import io.crate.types.TypeSignature;
+
+public class AllEqOperator extends AllOperator {
+
+    public static String NAME = OPERATOR_PREFIX + ComparisonExpression.Type.EQUAL.getValue();
+    public static Signature SIGNATURE = Signature.builder(AllEqOperator.NAME, FunctionType.SCALAR)
+            .argumentTypes(TypeSignature.parse("E"), TypeSignature.parse("array(E)"))
+            .returnType(Operator.RETURN_TYPE.getTypeSignature())
+            .typeVariableConstraints(typeVariable("E"))
+            .features(Feature.DETERMINISTIC, Feature.STRICTNULL)
+            .build();
+
+    public AllEqOperator(Signature signature, BoundSignature boundSignature, IntPredicate cmp) {
+        super(signature, boundSignature, cmp);
+    }
+
+    public static Query refMatchesAllArrayLiteral(Reference probe, List<?> values, LuceneQueryBuilder.Context context) {
+        // col = ALL ([1,2,3]) --> col=1 and col=2 and col=3
+        LinkedHashSet<?> uniqueValues = new LinkedHashSet<>(values);
+        if (uniqueValues.contains(null)) {
+            return new MatchNoDocsQuery("Cannot match nulls");
+        }
+        if (uniqueValues.isEmpty()) {
+            return new MatchNoDocsQuery("Cannot match unless there is at least one non-null candidate");
+        }
```

Comment:
```suggestion
        if (uniqueValues.isEmpty()) {
            return new MatchNoDocsQuery("Cannot match unless there is at least one non-null candidate");
        }
        if (uniqueValues.contains(null)) {
            return new MatchNoDocsQuery("Cannot match nulls");
        }
```

Minor, but if it is empty it implies that it doesn't contain `null`